tap_shopify/streams/metafields.py

```diff
@@ -64,115 +60,140 @@ def get_query_params(self, updated_at_min, updated_at_max, cursor=None):
             params["after"] = cursor
         return params
 
-    def get_resource_type_query(self, resource):
+    @staticmethod
+    def get_resource_type_query(resource):
         return {
             "customer": get_metafield_query_customers,
             "product": get_metafield_query_product,
             "collection": get_metafield_query_collection,
             "order": get_metafield_query_order,
-            "shop":get_metafield_query_shop
-            }.get(resource)
+            "shop": get_metafield_query_shop
+        }.get(resource)
 
     @shopify_error_handling
-    def call_api(self, query_params, query, data_key):
+    def call_api(self, query_params, query):
         response = shopify.GraphQL().execute(query=query, variables=query_params)
         response = json.loads(response)
         if "errors" in response.keys():
             raise ShopifyAPIError(response['errors'])
-        data = response.get("data", {}).get(data_key, {})
+        data = response.get("data", {})
         return data
 
-    def get_parents(self):
+    def transform_object(self, obj):
+        obj["value_type"] = obj["type"] or None
+        obj["updated_at"] = obj["updatedAt"]
+        if obj["value_type"] in ["json", "weight", "volume", "dimension", "rating"]:
+            value = obj.get("value")
+            try:
+                obj["value"] = json.loads(value) if value is not None else value
+            except json.decoder.JSONDecodeError:
+                LOGGER.info("Failed to decode JSON value for obj %s", obj.get('id'))
+                obj["value"] = value
+        return obj
+
+    def get_next_page_metafields(self, metafields, query_params, gql_resource):
+        owner_id = metafields.get("edges", [])[0]["node"]["owner"]["id"]
+        query = self.get_resource_type_query(gql_resource)()
```

Comment:
this is not the right way to use a static method with a self reference
please change the method implementation or update the function call.

Reply:
I have updated the function to the instance method.